staging/src/k8s.io/client-go/scale/client.go

```diff
@@ -213,3 +214,40 @@ func (c *namespacedScaleClient) Update(resource schema.GroupResource, scale *aut
 
 	return scaleObj.(*autoscaling.Scale), err
 }
+
+func (c *namespacedScaleClient) Patch(resource schema.GroupResource, name string, pt types.PatchType, data []byte) (*autoscaling.Scale, error) {
```

Comment:
I think this needs to take a full groupversionresource... the client isn't doing any conversion of the patch to the right version, so the caller needs to know the exact endpoint they are patching against

Reply:
The method would invoke `c.client.pathAndVersionFor` which returns the preferred version of given resource. Is it not desired?

Reply:
The patch that is passed in is for a specific version of auto scaling. It can't assume the incoming patch can be adapted to the discoverer scale version like update can

Reply:
fixed now!